### PR TITLE
Add pipline custom configuration doc for AWS

### PR DESCRIPTION
**Pull request by Jeffwan**
Add more instructions to use KFP in AWS.

### CODE REVIEW

**Comment by sarahmaddox** [comment]
/assign @joeliedtke 

Thanks @Jeffwan! I'm assigning this one to @joeliedtke as he's the tech writer for Pipelines.

**Comment by Jeffwan** [comment]
> /assign @joeliedtke
> 
> Thanks @Jeffwan! I'm assigning this one to @joeliedtke as he's the tech writer for Pipelines.

Thanks @sarahmaddox Sounds good

**Comment by joeliedtke** [comment]
Preview: https://deploy-preview-1852--competent-brattain-de2d6d.netlify.com/docs/aws/pipeline/

File: content/docs/aws/pipeline.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+Here's an example.
+<img src="/docs/images/aws/kfp-viewer-tensorboard.png"
+  alt="Kubeflow Pipeline viewer tensorboard"
+  class="mt-3 mb-3 border border-info rounded">
+
+## Support Tensorboard in Kubeflow Pipeline
+
+ [Tensorboard](/docs/pipelines/sdk/output-viewer/#tensorboard) need some extra settings on AWS.
+
+1. Create a Kubernetes secret `aws-secret` S3 read policies in `kubeflow` namespace, check above example.
```
Comment (joeliedtke) [inline content/docs/aws/pipeline.md]: policies in `kubeflow` namespace -> policies in the `kubeflow` namespace

End the sentence after 'namespace' and replace 'check above example' with a sentence describing where to find instructions.

File: content/docs/aws/pipeline.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+1. Create a configmap to store template spec of viewer tensorboard pod. Replace `<your_region>` with your S3 region.
```
Comment (joeliedtke) [inline content/docs/aws/pipeline.md]: Create a ConfigMap to store the configuration of TensorBoard on your cluster.

File: content/docs/aws/pipeline.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+          image: gcr.io/ml-pipeline/frontend:0.2.0
+          name: ml-pipeline-ui
+  ```
+
+Here's an example.
+<img src="/docs/images/aws/kfp-viewer-tensorboard.png"
+  alt="Kubeflow Pipeline viewer tensorboard"
+  class="mt-3 mb-3 border border-info rounded">
+
+## Support Tensorboard in Kubeflow Pipeline
```
[the remark this answers is not in the document]
Comment (joeliedtke) [reply content/docs/aws/pipeline.md]: Tensorboard -> TensorBoard

File: content/docs/aws/pipeline.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+ [Tensorboard](/docs/pipelines/sdk/output-viewer/#tensorboard) need some extra settings on AWS.
```
[the remark this answers is not in the document]
Comment (joeliedtke) [reply content/docs/aws/pipeline.md]: Tensorboard -> TensorBoard

File: content/docs/aws/pipeline.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+                              "value": "<your_region>"
+                          }
+                      ]
+                  }
+              ]
+          }
+      }
+  ```
+
+1. Update deployment `ml-pipeline-ui` to use this sepc by running `kubectl edit deployment ml-pipeline-ui -n kubeflow`.
```
Comment (joeliedtke) [inline content/docs/aws/pipeline.md]: Update the `ml-pipeline-ui` deployment to use the ConfigMap by running `kubectl edit deployment ml-pipeline-ui -n kubeflow`.

Is there anything else that people should know about running this command? How is the following YAML file related?

File: content/docs/aws/pipeline.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
++++
+title = "Configure Kubeflow Pipeline on AWS"
+weight = 90
```
[the remark this answers is not in the document]
Comment (Jeffwan) [reply content/docs/aws/pipeline.md]: Yeah. I will add some descriptions

File: content/docs/aws/pipeline.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
++++
+
+## S3 Access from Kubeflow Pipeline Pod
+
+If you write any files to S3 in your application, use `use_aws_secret` to attach aws secret to access S3.
```
[the remark this answers is not in the document]
Comment (Jeffwan) [reply content/docs/aws/pipeline.md]: This is KFP applications, not the control plane. User write files to S3 here and UI/Backend requires read access.

File: content/docs/aws/pipeline.md [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+        .....
+        output_artifact_paths={'mlpipeline-ui-metadata': '/mlpipeline-ui-metadata.json'}
+    )
+
+@dsl.pipeline(
+    name='IRIS Classification pipeline',
+    description='IRIS Classification using LR in SKLEARN'
+)
+def iris_pipeline():
+    iris_task = iris_comp().apply(use_aws_secret('aws-secret', 'AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY'))
```
[the remark this answers is not in the document]
Comment (Jeffwan) [reply content/docs/aws/pipeline.md]: This is much better. I will adopt your suggestion

File: content/docs/aws/pipeline.md [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+            name: config-volume
+        .....
+        volumes:
+        - configMap:
+            defaultMode: 420
+            name: ml-pipeline-ui-viewer-template
+          name: config-volume
+  ```
+
+### Replace KFP object storage with S3
```
[the remark this answers is not in the document]
Comment (Jeffwan) [reply content/docs/aws/pipeline.md]: Removed

File: content/docs/aws/pipeline.md [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+    iris_task = iris_comp().apply(use_aws_secret('aws-secret', 'AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY'))
+
+```
+
+## Support S3 Artifact Store
+
+Kubeflow Pipeline supports different artifact viewers. You can create files in S3 and reference them in output artifacts in your application like beflow.
+
+```python
+metadata = {
```
[the remark this answers is not in the document]
Comment (Jeffwan) [reply content/docs/aws/pipeline.md]: This is part of the application codes, I will add more code snippets here.

File: content/docs/aws/pipeline.md [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+          }
+        ]
+    }
+```
+
+In order for `ml-pipeline-ui` to read these artifacts,
+
+1. Create a Kubernetes secret `aws-secret` with at least S3 read policies in `kubeflow` namespace.
+
+  ```yaml
```
[the remark this answers is not in the document]
Comment (Jeffwan) [reply content/docs/aws/pipeline.md]: This is very clear! Thanks

File: content/docs/aws/pipeline.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Jeffwan) [reply content/docs/aws/pipeline.md]: All following points are the extra settings.

File: content/docs/aws/pipeline.md (hunk 5 above)
Comment (Jeffwan) [reply content/docs/aws/pipeline.md]: The following Yaml indicates the changes user need to make after running `edit`

File: content/docs/aws/pipeline.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (Jeffwan) [reply content/docs/aws/pipeline.md]: @joeliedtke I notice we have lots of references in following examples. It's not good to have these everywhere. I think I will have a separate section to create k8s Secret and mention place holder. For rest of examples, stick to one concrete example?

[the remark this answers is not in the document]
Comment (Jeffwan) [reply content/docs/aws/pipeline.md]: em. Actually I think we were not on the same page. `AWS_ACCESS_KEY_ID ` is not the credential value, it is a Key. User can definitely use different value but for most of the cases, we use `AWS_ACCESS_KEY_ID ` this fixed value

File: content/docs/aws/pipeline.md [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+              'type': 'tensorboard',
+              'source': s3://bucket/tb-events,
+          }
+        ]
+    }
+```
+
+In order for `ml-pipeline-ui` to read these artifacts,
+
+1. Create a Kubernetes secret `aws-secret` with at least S3 read policies in `kubeflow` namespace.
```
[the remark this answers is not in the document]
Comment (Jeffwan) [reply content/docs/aws/pipeline.md]: This section `S3 Access from Kubeflow Pipelines` will talk about creating credentials. All rest will have reference to this part.

**Comment by Jeffwan** [comment]
@joeliedtke Please have another review on this revision. Address most of the feedbacks besides this one. We can consider these two variables `AWS_ACCESS_KEY_ID ` and `AWS_SECRET_ACCESS_KEY` as fixed values. They are key name of value and widely used in other examples.

File: content/docs/aws/pipeline.md [hunk 13]
```diff
@@ -1,9 +1,34 @@
...
-title = "Configure Kubeflow Pipeline on AWS"
+title = "Configure Kubeflow Pipelines on AWS"
+description = "Customize Kubeflow Pipelines to use AWS Services"
 weight = 90
 +++
 
-## S3 Access from Kubeflow Pipeline Pod
+## S3 Access from Kubeflow Pipelines
+
+Currently, it's still recommended to use aws credentials or [kube2iam](https://github.com/jtblin/kube2iam) to managed S3 access from Kubeflow Pipelines. [IAM Role for Service Accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) requires applications to use latest AWS SDK to support `assume-web-identity-role`, we are still working on it. Please track progress in issue [#3405](https://github.com/kubeflow/pipelines/issues/3405)
```
Comment (joeliedtke) [inline content/docs/aws/pipeline.md:9]: Remove "please"

File: content/docs/aws/pipeline.md [hunk 14]
```diff
@@ -1,9 +1,34 @@
...
+
+A Kubernetes Secret is required by Kubeflow Pipelines and applications to access S3. Please make sure it has S3 read and write access.
```
Comment (joeliedtke) [inline content/docs/aws/pipeline.md:11]: Remove "please"

**Comment by Jeffwan** [comment]
@joeliedtke Address your latest comments. Please have another look

**Comment by joeliedtke** [comment]
/lgtm
/approve

Thanks for adding this @Jeffwan, this looks looks like it will be very helpful for people who want to use Kubeflow Pipelines on AWS.

**Pull request merged**